RealmModelGenerator/ObjectCContentGenerator.swift

```diff
@@ -0,0 +1,210 @@
+                } else {
+                    requiredProperties += ", @\"" + attr.name + "\""
+                }
+            }
+        }
+        
+        // Append relationship property
+        for relationship in entity.relationships {
+            hContent += "@property "
+            if relationship.isMany {
+                hContent += "RLMArray<" + (relationship.destination?.name)! + ">"
+            } else {
+                hContent += (relationship.destination?.name)!
+            }
+            
+            hContent += " * " + relationship.name + ";\n"
+        }
+        
+        hContent += "\n"
+        
+        appendPrimaryKey()
+        appendIndexedProperties(indexedProperties)
+        appendDefaultPropertyValues(defaultProperties)
+        appendIgnoredProperties(ignoredProperties)
+        appendRequiredProperties(requiredProperties)
+    }
+    
+    //MARK: Append attributes for property
+    func appendIndexedProperties(indexedProperties: String) {
+        if indexedProperties.isEmpty {
+            return
+        }
+        
+        mContent += "+ (NSArray *)indexedProperties {\n"
+        mContent += "\treturn @[" + indexedProperties + "];\n"
+        mContent += "}\n\n"
+    }
+    
+    //MARK: Append primary key
+    func appendPrimaryKey() {
+        let primaryKey = entity.primaryKey
+        if primaryKey == nil {
+            return
+        }
+        
+        mContent += "+(\((primaryKey?.type.objectCName)!) *)primaryKey {\n"
+        mContent += "\treturn @\"" + primaryKey!.name + "\";\n"
+        mContent += "}\n\n"
+    }
+
+    //MARK: Append default property value
+    func appendDefaultPropertyValues(defaultProperties: String) {
+        if defaultProperties.isEmpty {
+            return
+        }
+        
+        mContent += "+ (NSDictionary *)defaultPropertyValues {\n"
+        mContent += "\treturn @{" + defaultProperties + "};\n"
+        mContent += "}\n\n"
+    }
+    
+    //MARK: Append ignored properties
+    func appendIgnoredProperties(ignoredProperties: String) {
+        if ignoredProperties.isEmpty {
+            return
+        }
+        
+        mContent += "+ (NSArray *)ignoredProperties {\n"
+        mContent += "\treturn @[" + ignoredProperties + "];\n"
+        mContent += "}\n\n"
+    }
+    
+    //MARK: Append required properties
+    func appendRequiredProperties(requiredProperties: String) {
+        if requiredProperties.isEmpty {
+            return
+        }
+        
+        mContent += "+ (NSArray *)requiredProperties {\n"
+        mContent += "\treturn @[" + requiredProperties + "];\n"
+        mContent += "}\n\n"
+    }
+}
+
+extension AttributeType {
+    var objectCName:Swift.String {
+        get {
+            switch (self) {
+                case .Bool:
+                    return "BOOL"
+                case .Short:
+                    return "NSInteger"
+                case .Int:
+                    return "NSInteger"
+                case .Long:
```

Comment:
In realm object c doc, it use NSInteger for id in Primary Keys and Ignored Properties. In Cheatsheet, it use int ,float double for non-optional number and NSNumber<RLMInt> *, NSNumber<RLMFloat> * and NSNumber<RLMDouble> * for optional number. My solution is set isRequired(or isOptional) as input to get correct type.